pkg/services/folder/folderimpl/folder.go

```diff
@@ -249,6 +257,11 @@ func (s *Service) GetChildren(ctx context.Context, cmd *folder.GetChildrenQuery)
 		}
 	}
 
+	if len(filtered) < len(children) {
+		// add "shared with me" folder
+		filtered = append(filtered, &folder.SharedWithMeFolder)
+	}
```

Comment:
We only need to show "Shared" folder if user does not have access to all root folders. Otherwise, all folders are accessible because permissions are inherited.

Reply:
Ah, I see. And that can only be the case if we're listing children of the root (`cmd.UID == ""`). Thanks for explaining!